### DISCUSSION TITLE
MRTK3 deployment to HL2 issue

### DISCUSSION THREAD

**Discussion by kaveheng** [MixedRealityToolkit/MixedRealityToolkit-Unity#557] [Q&A] [answered]
I have developed a project based on MRTK3, Unity 2022.3.13f1, and Windows 10.
The issue is when I deployed it to HL2, I got the application in the menu, but when I started it nothing and after some seconds it went back to the splash image.
I also tried to deploy a sample scene from MRTKDevTemplate (same Unity version) and got the same result. 

What could be the issue? What am I doing wrong?

**Comment by anonymous2585** [comment, marked as answer]
Hi,
I got a simillar issue and lost a lot of time with it.

My issue was with the input system. It is known that the version 1.5.0 has a bug: https://learn.microsoft.com/en-us/windows/mixed-reality/develop/unity/known-issues (HL2 app crashes when hands are detected.)
But even when using version 1.5.1 or more, I still had exactly the problem you described (even with my hands in my back so  they aren't detected). One way to fix it was to force downgrade to input system 1.4.4.

I noticed that the version of the Visual Studio had an impact on my issue.
At first I compiled with Visual Studio 2019.6.4 (version forced by my company), and always had the same issue for every input system version above 1.4.4. I reported the bug to Unity, but they didn't care because they tested with an up-to-date VS 2019 version that doesn't have this issue.
I finally got VS 2022.4.4 from my company, and it solved this bug.

But I had another crash later in my app workflow, that I fixed by updating the c++ compiler from 14.36.32532 (default for VS 2022.4.4) to 14.37.32822.

Make sure to use an up-to-date VS 2019 or 2022.

Cheers

**Reply by kaveheng** [reply]
You've been a lifesaver! I've spent over a week experimenting with various solutions, installing multiple versions of Unity, and even starting from scratch using the empty template scene. The breakthrough came when I **updated** my **Visual Studio 2022** to the latest version as you hinted.  Indeed, in the new update, the C++ build function was enhanced. Now, everything is working smoothly. 
Thanks again for your help!
